byway/src/main/webapp/routepage.js

```diff
@@ -161,17 +154,36 @@ function getRouteOnload(){
   clearRoute();
   fetch('/api/stop')
   .then(response => response.json())
-  .then((trip) => {
+  .then(async (trip) => {
     if(trip != null){
-      start = end = trip.start;
-      destinations = trip.destinations;
-      route.push(...trip.route)
+      try{
+        let res = await findPlace(trip.start);
+        start = end = res.name;
+      } catch (error) {
+        console.error("Could not retrieve a start nor end point due to: ", error);
+      }
+      
+      for(let destination of trip.destinations){
```

Comment:
I think being a bit more descriptive with the variable names would help here. Since `trip.destinations` stores the place ids, consider renaming: `destination` -> `destinationId`. This goes for the `trip.route` list as well

Reply:
Yeah you're right, thanks for pointing it out. Changed.